test/Microsoft.ML.Tests/CSharpCodeGen.cs

```diff
@@ -2,24 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.ML.Runtime.RunTests;
 using Microsoft.ML.TestFramework;
 using System.IO;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.ML.Tests
 {
-    public class CSharpCodeGen : BaseTestClass
+    public class CSharpCodeGen : BaseTestBaseline
     {
         public CSharpCodeGen(ITestOutputHelper output) : base(output)
         {
         }
 
-        [Fact(Skip = "Temporary solution(Windows ONLY) to regenerate codegenerated CSharpAPI.cs")]
-        public void GenerateCSharpAPI()
+        [Fact]
+        public void TestGeneratedCSharpAPI()
         {
-            var cSharpAPIPath = Path.Combine(RootDir, @"src\\Microsoft.ML\\CSharpApi.cs");
-            Runtime.Tools.Maml.Main(new[] { $"? generator=cs{{csFilename={cSharpAPIPath}}}" });
+            var dataPath = GetOutputPath("Api.cs");
+            Runtime.Tools.Maml.Main(new[] { $"? generator=cs{{csFilename={dataPath}}}" });
+            
+            var basePath = GetDataPath("../../src/Microsoft.ML/CSharpApi.cs");
+            using (StreamReader baseline = OpenReader(basePath))
+            using (StreamReader result = OpenReader(dataPath))
+            {
+                for (; ; )
+                {
+                    string line1 = baseline.ReadLine();
+                    string line2 = result.ReadLine();
+
+                    if (line1 == null && line2 == null)
+                        break;
+                    if (line2 != null && line2.Contains(dataPath))
```

Comment:
What does this line test?

Reply:
Freshness of  CsharpApi.cs. It prevents you checkin code which contains stale CsharpApi.cs

Reply:
Just for info: In which case, this will pass/fail?

Reply:
If you made changes across solution  and you haven't update CSharpApi and your changes actually change CSharpApi it will fail.
If your changes haven't changed CSharpApi, it will pass. 
Basically I don't want people to check in stale or manually edited CSharpApi

Reply:
Why do you check `line2.Contains(dataPath)`? 
Why is the comparison with `dataPath`?

Reply:
oh, yes, artifacts of the past, which I shouldn't copy.

Reply:
Lol...I took it seriously....:)